treeherder/etl/jobs.py

```diff
@@ -207,24 +182,20 @@ def _load_job(repository, job_datum, push_id, lower_tier_signatures):
             'option_collection_hash': option_collection_hash
         })
 
-    if created:
-        # A new ReferenceDataSignature has been added, so we need
-        # to reload lower tier exclusions
-        lower_tier_signatures = _get_lower_tier_signatures(repository)
-
     tier = job_datum.get('tier') or 1
-    # job tier signatures override the setting from the job structure
-    # Check the signatures list for any supported lower tiers that have
-    # an active exclusion profile.
 
     result = job_datum.get('result', 'unknown')
 
+    # job tier signatures override the setting from the job structure
+    # Check the signatures list for any supported lower tiers that have
+    # an active exclusion profile.
     # As stated elsewhere, a job will end up in the lowest tier where its
     # signature belongs.  So if a signature is in Tier-2 and Tier-3, it
```

Comment:
This comment is no longer completely accurate, since the signature won't end up in the lowest tier, but whatever tier was defined last, which at the moment happens to be tier 3, due to the ordering of `TIER_SIGNATURES`.

That said, I wonder if support for a signature appearing in multiple tiers should just be removed, and `TIER_SIGNATURES` flattened, to simply the lookup here?

eg:
```python
TIER_MAPPINGS = [
    # mozilla-central
    '49f148889483d2b918968dc58a3dc128e0cf3bad': 2,
    'eca4450a7589f585abb6417e2ec50ec9f9222f30': 2,
    'a639fc8e9c851c23d8757ca4cc0bdf4f47191a8d': 2,
    'ece1657e1df7a7eccfad512ec42ca95c737c9967': 2,
    'd5d4966a6291e0d5038feb53706ce94815f2940f': 2,
    '168c3f4f8b6e86fa6c72a90a7594ca740773754a': 2,
    '68a10491197ebdb2e1fe7421948602e610756648': 2,
    '42e9a1ff1501fbd9c5bfc4ee2db3014ef9e03764': 2,
    'a4cc9dc681d9dee8d2b2e12473bd108e567f9a6e': 2,
    'd727b862b1e0d15051c2cf70d433b98405b9d24e': 2,
    '7910e346829fdaf579fba02e7bdd5bdb484b3328': 2,
    # mozilla-esr52
    '4226b6b3015e09de57c2d553584eba5057b7425e': 3,
    '4df5d64f9501123193b0b3cf4d2e9b3c84bc2076': 3,
    # mozilla-release
    '285c5e0ad301703887bde886e4fb539e34678692': 3,
    'effdff3f7ec01e57e53e97cdda9e3bbf0f243e44': 3,
]
```

and then:
```python
if lower_tier_signatures and signature_hash in lower_tier_signatures:
    tier = lower_tier_signatures[signature_hash]
```

Reply:
Yeah, this is a good idea, I'll go with this.  A much faster lookup, along with being simpler.  :)